docs/src/details/xml/tools/cli.rst

```diff
@@ -1,23 +1,163 @@
-
 .. _editor_cli:
 
 ===
 CLI
 ===
 
+The LVGL Editor provides a Node-based command line tool, ``lved-cli.js``, for generating/compiling code from XML, validating projects, syncing from Figma, comparing trees, and running headless UI tests.
+
+.. note::
+   Pre-release build. Intended for development and CI. Expires end of 2025.
+
 Overview
 ********
 
-Usage
-*****
+- **Binary:** ``lved-cli.js`` (Node script) `Download Here <https://github.com/lvgl/lvgl_editor/releases>`_
+- **Platform:** Node 18+ recommended
+- **Container engine:** Podman (if not on Windows, `Download Here <https://github.com/containers/podman/releases>`_)
+
+
+Quick start
+***********
+
+Generate C and H files from XMLs:
+
+.. code-block:: bash
+
+   lved-cli.js generate path/to/project
+
+Compile (web or node) the runtime Binary for previewing or testing the UI:
+
+.. code-block:: bash
+
+   lved-cli.js compile path/to/project --target web --start-service
+
+Validate content of the XML files (limit shown errors):
+
+.. code-block:: bash
+
+   lved-cli.js validate path/to/project --errorlimit 25
+
+Run the `<test>`(s) from the XML files 
+
+.. code-block:: bash
+
+   lved-cli.js run-all-tests path/to/project
+   lved-cli.js run-test path/to/project tests/test-file.xml
+
+Synchronize the styles and images from Figma: 
+
+.. code-block:: bash
+
+   lved-cli.js figma-sync path/to/project --start-service
+
+Commands
+********
+
+generate
+--------
+
+Generate code from XML.
+
+.. code-block:: bash
+
+   lved-cli.js generate <project-path> [--start-service]
+
+
+compile
+-------
+
+Compile for a target.
```

Comment:
We need more description as it's not obvious. Why someone wants to compile? What is the result? When is it needed? What can we do with it? Even I don't know.